r/tests/testthat/test-matrix_io.R

```diff
@@ -260,14 +264,24 @@ test_that("AnnData read backwards compatibility", {
     open_matrix_anndata_hdf5(file.path(dir, f), group="layers/transpose") %>%
       as("dgCMatrix") %>%
       expect_identical(ans)
+    open_matrix_anndata_hdf5(file.path(dir, f), group="layers/dense") %>%
+      as("dgCMatrix") %>%
```

Comment:
I think it would be good to check for writing matrices that were originally read dense.  I did a little bit of testing with `write_matrix_hdf5()`, and `write_matrix_memory()` and they appear to be working.  But it never hurts to have a check in case of a regression!

Reply:
Since `as("dgCMatrix")` calls `write_matrix_memory()` internally, I think this is already covered here